editor/src/core/model/performance-scripts.tsx

```diff
@@ -66,15 +74,120 @@ function measureStep(prefix: string, framesPassed: number): void {
   )
 }
 
+async function loadProject(
+  dispatch: DebugDispatch,
+  builtInDependencies: BuiltInDependencies,
+  projectContents: ProjectContentTreeRoot,
+): Promise<boolean> {
+  const persistentModel: PersistentModel = {
+    forkedFromProjectId: null,
+    projectVersion: CURRENT_PROJECT_VERSION,
+    projectDescription: 'Performance Test Project',
+    projectContents: projectContents,
+    exportsInfo: [],
+    lastUsedFont: null,
+    hiddenInstances: [],
+    codeEditorErrors: {
+      buildErrors: {},
+      lintErrors: {},
+    },
+    fileBrowser: {
+      minimised: false,
+    },
+    dependencyList: {
+      minimised: false,
+    },
+    projectSettings: {
+      minimised: false,
+    },
+    navigator: {
+      minimised: false,
+    },
+  }
+
+  // Load the project itself.
+  await load(dispatch, persistentModel, 'Test', '999999', builtInDependencies, false)
+
+  // Wait for the editor to stabilise, ensuring that the canvas can render for example.
+  const startWaitingTime = Date.now()
+  let editorReady: boolean = false
+  let itemSelected: boolean = false
+  let canvasPopulated: boolean = false
+  let codeEditorPopulated: boolean = false
+  let codeEditorLoaded: boolean = false
+  while (startWaitingTime + 20000 > Date.now() && !editorReady) {
```

Comment:
I would extract all the magic time numbers to named constants.

Reply:
Yeah, that's a good shout.

Reply:
Fixed in 2d46045d79166ca6ca771186e5192e00c42b06a0.